src/devices/RGBDSensorFromRosTopic/RGBDSensorFromRosTopic.cpp

```diff
@@ -0,0 +1,380 @@
+{
+    if (m_rgb_input_processor == nullptr)
+    {
+        horizontalFov=0;
+        verticalFov=0;
+        return true;
+    }
+    return m_rgb_input_processor->getFOV(horizontalFov, verticalFov);
+}
+
+bool RGBDSensorFromRosTopic::getRgbMirroring(bool& mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "Mirroring not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::setRgbMirroring(bool mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "Mirroring not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::getRgbIntrinsicParam(Property& intrinsic)
+{
+    if (m_rgb_input_processor == nullptr)
+    {
+        intrinsic.clear();
+        return  true;
+    }
+    return m_rgb_input_processor->getIntrinsicParam(intrinsic);
+}
+
+int  RGBDSensorFromRosTopic::getDepthHeight()
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        return  0;
+    }
+    return (int)m_depth_input_processor->getHeight();
+}
+
+int  RGBDSensorFromRosTopic::getDepthWidth()
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        return  0;
+    }
+    return (int)m_depth_input_processor->getWidth();
+}
+
+bool RGBDSensorFromRosTopic::getDepthFOV(double& horizontalFov, double& verticalFov)
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        horizontalFov = 0;
+        verticalFov = 0;
+        return true;
+    }
+    return m_depth_input_processor->getFOV(horizontalFov, verticalFov);
+}
+
+bool RGBDSensorFromRosTopic::getDepthIntrinsicParam(Property& intrinsic)
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        intrinsic.clear();
+        return true;
+    }
+    return m_depth_input_processor->getIntrinsicParam(intrinsic);
+}
+
+double RGBDSensorFromRosTopic::getDepthAccuracy()
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthAccuracy not supported";
+    return 0;
+}
+
+bool RGBDSensorFromRosTopic::getDepthClipPlanes(double& nearPlane, double& farPlane)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthClipPlanes not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::setDepthClipPlanes(double nearPlane, double farPlane)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "setDepthClipPlanes not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::getDepthMirroring(bool& mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthMirroring not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::setDepthMirroring(bool mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "setDepthMirroring not supported";
+    return false;
+}
+
+bool RGBDSensorFromRosTopic::getExtrinsicParam(Matrix& extrinsic)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getExtrinsicParam not supported";
+    return  false;
+};
+
+bool RGBDSensorFromRosTopic::getRgbImage(FlexImage& rgbImage, Stamp* timeStamp)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor!=nullptr)
+        bool rgb_ok = m_rgb_input_processor->getLastRGBData(rgbImage, *timeStamp);
```

Comment:
Codacy found an issue: [Variable 'rgb_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6096993)